server/src/test/java/io/crate/types/TypeConversionTest.java

```diff
@@ -97,6 +98,8 @@ public void numberConversionTest() throws Exception {
                 var t = DataTypes.fromId(id);
                 if (t.equals(DataTypes.IP)) {
                     byteVal = (byte) Math.abs(byteVal == Byte.MIN_VALUE ? byteVal >> 1 : byteVal);
+                } else if (t.equals(DataTypes.TIMETZ)) {
```

Comment:
These additions here shouldn't be necessary anymore, since TIMETZ is no longer part of the allowed conversions for BYTE.